lib/srv/ctx.go

```diff
@@ -449,28 +450,18 @@ func (c *ServerContext) AddCloser(closer io.Closer) {
 
 // GetAgent returns a agent.Agent which represents the capabilities of an SSH agent.
 func (c *ServerContext) GetAgent() agent.Agent {
-	c.RLock()
-	defer c.RUnlock()
-	return c.agent
+	if c.Parent == nil {
+		return nil
+	}
```

Comment:
Shouldn't `c.Parent` always be set?

Reply:
Removed `nil` checks for methods which modify state.  All the getters on the zero value of `srv.ConnectionContext` return zero values themselves instead of panicking (a fact that some tests rely on), so I kept the `nil` checks on the getters in order to preserve this behavior.